genesyscloud/user/resource_genesyscloud_user.go

```diff
@@ -0,0 +1,287 @@
+package user
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"terraform-provider-genesyscloud/genesyscloud/util/constants"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v130/platformclientv2"
+)
+
+type AgentUtilizationWithLabels struct {
+	Utilization       map[string]util.MediaUtilization `json:"utilization"`
+	LabelUtilizations map[string]util.LabelUtilization `json:"labelUtilizations"`
+	Level             string                           `json:"level"`
+}
+
+func GetAllUsers(ctx context.Context, sdkConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	proxy := getUserProxy(sdkConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	// Newly created resources often aren't returned unless there's a delay
+	time.Sleep(5 * time.Second)
+
+	users, proxyResponse, err := proxy.getAllUser(ctx)
+	if err != nil {
+		return nil, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to get page of users error: %s", err), proxyResponse)
+	}
+
+	// Add resources to metamap
+	for _, user := range *users {
+		resources[*user.Id] = &resourceExporter.ResourceMeta{Name: *user.Email}
+	}
+
+	return resources, nil
+}
+
+func createUser(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getUserProxy(sdkConfig)
+
+	email := d.Get("email").(string)
+	name := d.Get("name").(string)
+	password := d.Get("password").(string)
+	state := d.Get("state").(string)
+	divisionID := d.Get("division_id").(string)
+	department := d.Get("department").(string)
+	title := d.Get("title").(string)
+	manager := d.Get("manager").(string)
+	acdAutoAnswer := d.Get("acd_auto_answer").(bool)
+
+	addresses, addrErr := buildSdkAddresses(d)
+	if addrErr != nil {
+		return addrErr
+	}
+
+	// Check for a deleted user before creating
+	id, _ := getDeletedUserId(email, proxy)
+	if id != nil {
+		d.SetId(*id)
+		return restoreDeletedUser(ctx, d, meta, proxy)
+	}
+
+	createUser := platformclientv2.Createuser{
+		Email:      &email,
```

Comment:
Instead of creating variables and then passing in the address of the variables you can do this.
```
Email: platformclientv2.String(d.Get("email").(string))
```
This is assuming the attribute is required. There's a lot of code in the resource so lets try to shorten things as much as we can

Reply:
Email anyway cannot be used as you suggested as it is being used a argument to getDeletedUserId(), however i update other places wherever possible. thanks.